### PR TITLE
list campaign channel tests - initial work

### PR DESCRIPTION
**Pull request by KaliedaRik**
## What does this change?
List all the tests associated with a campaign. Each test listed includes a link to that test in RRCP

This PR covers the basic functionality of listing all tests associated with a campaign, presented by channels. 

Each test displays in a card with basic information (status, priority, variant names, targeted cohorts and regions) visualised in a way that should allow users a quick way to keep check on their work while creating or managing a campaign.

Each card also includes functionality to show errors such as no variants being created for a test, or no regions selected. The test name is also a link to take the user directly to the test page where it can be edited and managed.

### Screenshots

Screenshot 2022-06-28 at 14 09 50

Screenshot 2022-06-28 at 14 10 16

### CODE REVIEW

File: public/src/components/channelManagement/campaigns/CampaignsEditor.tsx [hunk 1]
```diff
@@ -18,27 +21,135 @@ const useStyles = makeStyles(({ spacing, palette }: Theme) => ({
...
+    fetchCampaignTests(name).then(tests => {
+      // sort by priority
+      const sortedTests = tests.sort((a: CombinedTest, b: CombinedTest) => {
+        return a.priority - b.priority;
+      });
+      setTestData(sortedTests);
+    });
+  }, [campaign]);
+
+  console.log(testData);
```
Comment (KaliedaRik) [inline public/src/components/channelManagement/campaigns/CampaignsEditor.tsx:122]: The console.log line needs to go

File: public/src/components/channelManagement/campaigns/ChannelCard.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        tests.map(test => (
+          <TestCard
+            test={test}
+            keyId={`${getKey(test)}_LINK`}
+            linkPath={channelData.link}
+            key={`${getKey(test)}_CARD`}
+          />
+        ))
+      ) : (
+        <div className={classes.noTestsWarning}>No tests have been set up for this channel</div>
```
Comment (MaelGNM) [inline public/src/components/channelManagement/campaigns/ChannelCard.tsx:47]: Pretty minor thing but the warning messages are not quite consistent in terms of punctuation and case:
`No tests have been set up for this channel`
`No variants have been created for this test!`
`No locations have been selected for this Test`

File: public/src/components/channelManagement/campaigns/CampaignsForm.tsx [hunk 3]
```diff
@@ -46,6 +49,19 @@ const useStyles = makeStyles(({ spacing, typography }: Theme) => ({
   },
 }));
 
+export type CombinedVariant = BannerVariant | EpicVariant | HeaderVariant;
+
+interface TestEnhancements {
```
Comment (tomrf1) [inline public/src/components/channelManagement/campaigns/CampaignsForm.tsx:54]: I would just add these fields to the `Test` model in `shared.ts`. Then you can just use `Test` and remove `TestEnhancements`, `CombinedTest`, `EnhancedBannerTest` etc

File: public/src/utils/requests.tsx [hunk 4]
```diff
@@ -108,6 +108,13 @@ export function fetchSupportFrontendSettings(
   return fetchSettings(`/support-frontend/${settingsType}`);
 }
 
+export function fetchCampaignTests(
+  campaign: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tomrf1) [inline public/src/utils/requests.tsx:113]: If you return `Promise<Test[]>` here then the eslint-disable isn't necessary

File: public/src/components/channelManagement/campaigns/CampaignsEditor.tsx (hunk 1 above)
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsEditor.tsx:122]: Fixed

File: public/src/components/channelManagement/campaigns/ChannelCard.tsx (hunk 2 above)
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/ChannelCard.tsx:47]: Fixed:
`No Tests have been set up for this Channel.`
`No Variants have been created for this Test.`
`No locations have been selected for this Test.`

File: public/src/components/channelManagement/campaigns/CampaignsForm.tsx (hunk 3 above)
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/CampaignsForm.tsx:54]: Took a while, but ... done!

File: public/src/utils/requests.tsx (hunk 4 above)
Comment (KaliedaRik) [reply public/src/utils/requests.tsx:113]: Done

File: public/src/components/channelManagement/campaigns/TestCard.tsx [hunk 5]
```diff
@@ -84,24 +84,24 @@ const useStyles = makeStyles(({ spacing }: Theme) => ({
...
 
-  const getVariantNames = (variants: CombinedVariant[]) => {
+  const getVariantNames = (variants: Variant[]) => {
     if (variants.length > 0) {
       return (
         <div className={classes.variantsData}>Variants: {variants.map(v => v.name).join(', ')}</div>
       );
     }
-    return <div className={classes.dataWarning}>No variants have been created for this test!</div>;
+    return <div className={classes.dataWarning}>No Variants have been created for this Test.</div>;
```
Comment (tomrf1) [inline public/src/components/channelManagement/campaigns/TestCard.tsx:101]: was this accidentally changed from `No variants have been created for this test!`?

**Review by tomrf1 — APPROVED** [review]
looks great!

File: public/src/components/channelManagement/campaigns/ChannelCard.tsx [hunk 6]
```diff
@@ -44,7 +44,7 @@ function ChannelCard({ channelData, tests }: ChannelCardProps): React.ReactEleme
           />
         ))
       ) : (
-        <div className={classes.noTestsWarning}>No tests have been set up for this channel</div>
+        <div className={classes.noTestsWarning}>No Tests have been set up for this Channel.</div>
```
[the remark this answers is not in the document]
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/ChannelCard.tsx:47]: No - deliberate, following on from Mael's suggestion.

File: public/src/components/channelManagement/campaigns/TestCard.tsx (hunk 5 above)
Comment (KaliedaRik) [reply public/src/components/channelManagement/campaigns/TestCard.tsx:101]: Again - deliberate, following on from Mael's suggestion.

**Pull request merged**